Test Fails on M1 Mac

**What happened**:

I tried running `make unit` on an M1 Mac and was running in the failure below:

```bash
--- FAIL: TestAllLayersResolver_AllLocations (0.03s)
    image_all_layers_resolver_test.go:822: 
                Error Trace:    /Users/henrysachs/projects/temp/syft/syft/source/image_all_layers_resolver_test.go:822
                Error:          elements differ
                            
                                extra elements in list A:
                                ([]interface {}) (len=2) {
                                 (string) (len=24) "/lib/ld-musl-x86_64.so.1",
                                 (string) (len=26) "/lib/libc.musl-x86_64.so.1"
                                }
                            
                            
                                extra elements in list B:
                                ([]interface {}) (len=2) {
                                 (string) (len=25) "/lib/ld-musl-aarch64.so.1",
                                 (string) (len=27) "/lib/libc.musl-aarch64.so.1"
                                }
                            
                            
                                listA:
                                ([]string) (len=112) {
                                 (string) (len=11) "/Dockerfile",
                                 (string) (len=11) "/file-1.txt",
                                 (string) (len=11) "/file-3.txt",
                                 (string) (len=7) "/target",
                                 (string) (len=18) "/target/file-2.txt",
                                 (string) (len=8) "/.wh.bin",
                                 (string) (len=15) "/.wh.file-1.txt",
                                 (string) (len=8) "/.wh.lib",
                                 (string) (len=4) "/bin",
                                 (string) (len=9) "/bin/arch",
                                 (string) (len=8) "/bin/ash",
                                 (string) (len=11) "/bin/base64",
                                 (string) (len=13) "/bin/bbconfig",
                                 (string) (len=12) "/bin/busybox",
                                 (string) (len=8) "/bin/cat",
                                 (string) (len=11) "/bin/chattr",
                                 (string) (len=10) "/bin/chgrp",
                                 (string) (len=10) "/bin/chmod",
                                 (string) (len=10) "/bin/chown",
                                 (string) (len=7) "/bin/cp",
                                 (string) (len=9) "/bin/date",
                                 (string) (len=7) "/bin/dd",
                                 (string) (len=7) "/bin/df",
                                 (string) (len=10) "/bin/dmesg",
                                 (string) (len=18) "/bin/dnsdomainname",
                                 (string) (len=13) "/bin/dumpkmap",
                                 (string) (len=9) "/bin/echo",
                                 (string) (len=7) "/bin/ed",
                                 (string) (len=10) "/bin/egrep",
                                 (string) (len=10) "/bin/false",
                                 (string) (len=12) "/bin/fatattr",
                                 (string) (len=12) "/bin/fdflush",
                                 (string) (len=10) "/bin/fgrep",
                                 (string) (len=10) "/bin/fsync",
                                 (string) (len=11) "/bin/getopt",
                                 (string) (len=9) "/bin/grep",
                                 (string) (len=11) "/bin/gunzip",
                                 (string) (len=9) "/bin/gzip",
                                 (string) (len=13) "/bin/hostname",
                                 (string) (len=11) "/bin/ionice",
                                 (string) (len=11) "/bin/iostat",
                                 (string) (len=11) "/bin/ipcalc",
                                 (string) (len=13) "/bin/kbd_mode",
                                 (string) (len=9) "/bin/kill",
                                 (string) (len=9) "/bin/link",
                                 (string) (len=12) "/bin/linux32",
                                 (string) (len=12) "/bin/linux64",
                                 (string) (len=7) "/bin/ln",
                                 (string) (len=10) "/bin/login",
                                 (string) (len=7) "/bin/ls",
                                 (string) (len=11) "/bin/lsattr",
                                 (string) (len=9) "/bin/lzop",
                                 (string) (len=13) "/bin/makemime",
                                 (string) (len=10) "/bin/mkdir",
                                 (string) (len=10) "/bin/mknod",
                                 (string) (len=11) "/bin/mktemp",
                                 (string) (len=9) "/bin/more",
                                 (string) (len=10) "/bin/mount",
                                 (string) (len=15) "/bin/mountpoint",
                                 (string) (len=11) "/bin/mpstat",
                                 (string) (len=7) "/bin/mv",
                                 (string) (len=12) "/bin/netstat",
                                 (string) (len=9) "/bin/nice",
                                 (string) (len=10) "/bin/pidof",
                                 (string) (len=9) "/bin/ping",
                                 (string) (len=10) "/bin/ping6",
                                 (string) (len=18) "/bin/pipe_progress",
                                 (string) (len=13) "/bin/printenv",
                                 (string) (len=7) "/bin/ps",
                                 (string) (len=8) "/bin/pwd",
                                 (string) (len=14) "/bin/reformime",
                                 (string) (len=8) "/bin/rev",
                                 (string) (len=7) "/bin/rm",
                                 (string) (len=10) "/bin/rmdir",
                                 (string) (len=14) "/bin/run-parts",
                                 (string) (len=8) "/bin/sed",
                                 (string) (len=12) "/bin/setpriv",
                                 (string) (len=14) "/bin/setserial",
                                 (string) (len=7) "/bin/sh",
                                 (string) (len=10) "/bin/sleep",
                                 (string) (len=9) "/bin/stat",
                                 (string) (len=9) "/bin/stty",
                                 (string) (len=7) "/bin/su",
                                 (string) (len=9) "/bin/sync",
                                 (string) (len=8) "/bin/tar",
                                 (string) (len=10) "/bin/touch",
                                 (string) (len=9) "/bin/true",
                                 (string) (len=11) "/bin/umount",
                                 (string) (len=10) "/bin/uname",
                                 (string) (len=11) "/bin/usleep",
                                 (string) (len=10) "/bin/watch",
                                 (string) (len=9) "/bin/zcat",
                                 (string) (len=4) "/lib",
                                 (string) (len=8) "/lib/apk",
                                 (string) (len=11) "/lib/apk/db",
                                 (string) (len=21) "/lib/apk/db/installed",
                                 (string) (len=16) "/lib/apk/db/lock",
                                 (string) (len=23) "/lib/apk/db/scripts.tar",
                                 (string) (len=20) "/lib/apk/db/triggers",
                                 (string) (len=13) "/lib/apk/exec",
                                 (string) (len=13) "/lib/firmware",
                                 (string) (len=24) "/lib/ld-musl-x86_64.so.1",
                                 (string) (len=21) "/lib/libapk.so.3.12.0",
                                 (string) (len=26) "/lib/libc.musl-x86_64.so.1",
                                 (string) (len=19) "/lib/libcrypto.so.3",
                                 (string) (len=16) "/lib/libssl.so.3",
                                 (string) (len=14) "/lib/libz.so.1",
                                 (string) (len=19) "/lib/libz.so.1.2.13",
                                 (string) (len=9) "/lib/mdev",
                                 (string) (len=19) "/lib/modules-load.d",
                                 (string) (len=13) "/lib/sysctl.d",
                                 (string) (len=28) "/lib/sysctl.d/00-alpine.conf"
                                }
                            
                            
                                listB:
                                ([]string) (len=112) {
                                 (string) (len=8) "/.wh.bin",
                                 (string) (len=15) "/.wh.file-1.txt",
                                 (string) (len=8) "/.wh.lib",
                                 (string) (len=11) "/Dockerfile",
                                 (string) (len=4) "/bin",
                                 (string) (len=9) "/bin/arch",
                                 (string) (len=8) "/bin/ash",
                                 (string) (len=11) "/bin/base64",
                                 (string) (len=13) "/bin/bbconfig",
                                 (string) (len=12) "/bin/busybox",
                                 (string) (len=8) "/bin/cat",
                                 (string) (len=11) "/bin/chattr",
                                 (string) (len=10) "/bin/chgrp",
                                 (string) (len=10) "/bin/chmod",
                                 (string) (len=10) "/bin/chown",
                                 (string) (len=7) "/bin/cp",
                                 (string) (len=9) "/bin/date",
                                 (string) (len=7) "/bin/dd",
                                 (string) (len=7) "/bin/df",
                                 (string) (len=10) "/bin/dmesg",
                                 (string) (len=18) "/bin/dnsdomainname",
                                 (string) (len=13) "/bin/dumpkmap",
                                 (string) (len=9) "/bin/echo",
                                 (string) (len=7) "/bin/ed",
                                 (string) (len=10) "/bin/egrep",
                                 (string) (len=10) "/bin/false",
                                 (string) (len=12) "/bin/fatattr",
                                 (string) (len=12) "/bin/fdflush",
                                 (string) (len=10) "/bin/fgrep",
                                 (string) (len=10) "/bin/fsync",
                                 (string) (len=11) "/bin/getopt",
                                 (string) (len=9) "/bin/grep",
                                 (string) (len=11) "/bin/gunzip",
                                 (string) (len=9) "/bin/gzip",
                                 (string) (len=13) "/bin/hostname",
                                 (string) (len=11) "/bin/ionice",
                                 (string) (len=11) "/bin/iostat",
                                 (string) (len=11) "/bin/ipcalc",
                                 (string) (len=13) "/bin/kbd_mode",
                                 (string) (len=9) "/bin/kill",
                                 (string) (len=9) "/bin/link",
                                 (string) (len=12) "/bin/linux32",
                                 (string) (len=12) "/bin/linux64",
                                 (string) (len=7) "/bin/ln",
                                 (string) (len=10) "/bin/login",
                                 (string) (len=7) "/bin/ls",
                                 (string) (len=11) "/bin/lsattr",
                                 (string) (len=9) "/bin/lzop",
                                 (string) (len=13) "/bin/makemime",
                                 (string) (len=10) "/bin/mkdir",
                                 (string) (len=10) "/bin/mknod",
                                 (string) (len=11) "/bin/mktemp",
                                 (string) (len=9) "/bin/more",
                                 (string) (len=10) "/bin/mount",
                                 (string) (len=15) "/bin/mountpoint",
                                 (string) (len=11) "/bin/mpstat",
                                 (string) (len=7) "/bin/mv",
                                 (string) (len=12) "/bin/netstat",
                                 (string) (len=9) "/bin/nice",
                                 (string) (len=10) "/bin/pidof",
                                 (string) (len=9) "/bin/ping",
                                 (string) (len=10) "/bin/ping6",
                                 (string) (len=18) "/bin/pipe_progress",
                                 (string) (len=13) "/bin/printenv",
                                 (string) (len=7) "/bin/ps",
                                 (string) (len=8) "/bin/pwd",
                                 (string) (len=14) "/bin/reformime",
                                 (string) (len=8) "/bin/rev",
                                 (string) (len=7) "/bin/rm",
                                 (string) (len=10) "/bin/rmdir",
                                 (string) (len=14) "/bin/run-parts",
                                 (string) (len=8) "/bin/sed",
                                 (string) (len=12) "/bin/setpriv",
                                 (string) (len=14) "/bin/setserial",
                                 (string) (len=7) "/bin/sh",
                                 (string) (len=10) "/bin/sleep",
                                 (string) (len=9) "/bin/stat",
                                 (string) (len=9) "/bin/stty",
                                 (string) (len=7) "/bin/su",
                                 (string) (len=9) "/bin/sync",
                                 (string) (len=8) "/bin/tar",
                                 (string) (len=10) "/bin/touch",
                                 (string) (len=9) "/bin/true",
                                 (string) (len=11) "/bin/umount",
                                 (string) (len=10) "/bin/uname",
                                 (string) (len=11) "/bin/usleep",
                                 (string) (len=10) "/bin/watch",
                                 (string) (len=9) "/bin/zcat",
                                 (string) (len=11) "/file-1.txt",
                                 (string) (len=11) "/file-3.txt",
                                 (string) (len=4) "/lib",
                                 (string) (len=8) "/lib/apk",
                                 (string) (len=11) "/lib/apk/db",
                                 (string) (len=21) "/lib/apk/db/installed",
                                 (string) (len=16) "/lib/apk/db/lock",
                                 (string) (len=23) "/lib/apk/db/scripts.tar",
                                 (string) (len=20) "/lib/apk/db/triggers",
                                 (string) (len=13) "/lib/apk/exec",
                                 (string) (len=13) "/lib/firmware",
                                 (string) (len=25) "/lib/ld-musl-aarch64.so.1",
                                 (string) (len=21) "/lib/libapk.so.3.12.0",
                                 (string) (len=27) "/lib/libc.musl-aarch64.so.1",
                                 (string) (len=19) "/lib/libcrypto.so.3",
                                 (string) (len=16) "/lib/libssl.so.3",
                                 (string) (len=14) "/lib/libz.so.1",
                                 (string) (len=19) "/lib/libz.so.1.2.13",
                                 (string) (len=9) "/lib/mdev",
                                 (string) (len=19) "/lib/modules-load.d",
                                 (string) (len=13) "/lib/sysctl.d",
                                 (string) (len=28) "/lib/sysctl.d/00-alpine.conf",
                                 (string) (len=7) "/target",
                                 (string) (len=18) "/target/file-2.txt"
                                }
                Test:           TestAllLayersResolver_AllLocations
                Messages:       expected all paths to be indexed, but found different paths: 
                                  []string{
                                -       "/Dockerfile",
                                +       "/bin/uname",
                                -       "/file-1.txt",
                                +       "/bin/ping6",
                                -       "/file-3.txt",
                                -       "/target",
                                -       "/target/file-2.txt",
                                -       "/.wh.bin",
                                -       "/.wh.file-1.txt",
                                -       "/.wh.lib",
                                -       "/bin",
                                -       "/bin/arch",
                                        "/bin/ash",
                                -       "/bin/base64",
                                +       "/lib/libcrypto.so.3",
                                -       "/bin/bbconfig",
                                -       "/bin/busybox",
                                -       "/bin/cat",
                                -       "/bin/chattr",
                                -       "/bin/chgrp",
                                -       "/bin/chmod",
                                -       "/bin/chown",
                                -       "/bin/cp",
                                -       "/bin/date",
                                -       "/bin/dd",
                                -       "/bin/df",
                                -       "/bin/dmesg",
                                -       "/bin/dnsdomainname",
                                        ... // 10 identical, 21 removed, 42 inserted, and 56 modified elements
                                  }
FAIL
        github.com/anchore/syft/syft/source     coverage: 67.0% of statements
FAIL    github.com/anchore/syft/syft/source     6.666s
FAIL
make: *** [unit] Error 1
```

**What you expected to happen**:

**Steps to reproduce the issue**:

**Anything else we need to know?**:

**Environment**:
- Output of `syft version`:
- OS (e.g: `cat /etc/os-release` or similar):

--- a/syft/source/image_all_layers_resolver_test.go
+++ b/syft/source/image_all_layers_resolver_test.go
@@ -1,7 +1,9 @@
 package source
 
 import (
+	"fmt"
 	"io"
+	"runtime"
 	"sort"
 	"testing"
 
@@ -689,6 +691,11 @@ func Test_imageAllLayersResolver_resolvesLinks(t *testing.T) {
 func TestAllLayersResolver_AllLocations(t *testing.T) {
 	img := imagetest.GetFixtureImage(t, "docker-archive", "image-files-deleted")
 
+	arch := "x86_64"
+	if runtime.GOARCH == "arm64" {
+		arch = "aarch64"
+	}
+
 	resolver, err := newAllLayersResolver(img)
 	assert.NoError(t, err)
 
@@ -799,9 +806,9 @@ func TestAllLayersResolver_AllLocations(t *testing.T) {
 		"/lib/apk/db/triggers",
 		"/lib/apk/exec",
 		"/lib/firmware",
-		"/lib/ld-musl-x86_64.so.1",
+		fmt.Sprintf("/lib/ld-musl-%s.so.1", arch),
 		"/lib/libapk.so.3.12.0",
-		"/lib/libc.musl-x86_64.so.1",
+		fmt.Sprintf("/lib/libc.musl-%s.so.1", arch),
 		"/lib/libcrypto.so.3",
 		"/lib/libssl.so.3",
 		"/lib/libz.so.1",

